Set verbatimModuleSyntax to true

Pretty much #1044, but for the newer, slightly stricter flag.

## Sorting

- **I'm submitting a ...**

  - [ ] bug report
  - [x] feature request
  - [ ] support request

- I confirm that I
  - [x] used the [search](https://github.com/lukeautry/tsoa/search?type=Issues) to make sure that a similar issue hasn't already been submit

## Expected Behavior

```ts
// routes.ts
import type { FieldErrors, TsoaRoute, HttpStatusCodeLiteral, TsoaResponse } from '@tsoa/runtime';
import { Controller, ValidationService, ValidateError, fetchMiddlewares } from '@tsoa/runtime';
```

## Current Behavior

```ts
// routes.ts
import { Controller, ValidationService, FieldErrors, ValidateError, TsoaRoute, HttpStatusCodeLiteral, TsoaResponse, fetchMiddlewares } from '@tsoa/runtime';
```

## Possible Solution

Changing this line:

https://github.com/lukeautry/tsoa/blob/master/packages/cli/src/routeGeneration/templates/express.hbs#L4
https://github.com/lukeautry/tsoa/blob/master/packages/cli/src/routeGeneration/templates/hapi.hbs#L4
https://github.com/lukeautry/tsoa/blob/master/packages/cli/src/routeGeneration/templates/koa.hbs#L5

## Detailed Description

This stricter version requries *all* types to be explicitly imported as such. With `importsNotUsedAsValues` you could include type imports with your "normal" imports, but with `verbatimModuleSyntax`, this is no longer allowed.

For example:

```ts
// not allowed by neither `importsNotUsedAsValues` nor `verbatimModuleSyntax`
import { MyInterface } from 'whatever';
```

```ts
// fine with `importsNotUsedAsValues`, not allowed by `verbatimModuleSyntax`
import { someFunction, MyInterface } from 'whatever';
```

```ts
// fine with both
import { someFunction, type MyInterface } from 'whatever';
```

```ts
// fine with both
import { someFunction } from 'whatever';
import type { MyInterface } from 'whatever';
```

So just like #1044 before, this will require a small tweak to the handlebars template(s).

## Breaking change?

No

I'm still seeing this issue in version 6.4.0. 

```bash
$ tsc --noEmit
build/routes.ts:4:10 - error TS1484: 'TsoaRoute' is a type and must be imported using a type-only import when 'verbatimModuleSyntax' is enabled.

4 import { TsoaRoute, fetchMiddlewares, ExpressTemplateService } from '@tsoa/runtime';
           ~~~~~~~~~


Found 1 error in build/routes.ts:4
```

--- a/packages/cli/src/routeGeneration/templates/express.hbs
+++ b/packages/cli/src/routeGeneration/templates/express.hbs
@@ -1,7 +1,8 @@
 /* tslint:disable */
 /* eslint-disable */
 // WARNING: This file was auto-generated with tsoa. Please do not modify it. Re-run tsoa to re-generate this file: https://github.com/lukeautry/tsoa
-import { TsoaRoute, fetchMiddlewares, ExpressTemplateService } from '@tsoa/runtime';
+import type { TsoaRoute } from '@tsoa/runtime';
+import {  fetchMiddlewares, ExpressTemplateService } from '@tsoa/runtime';
 {{#each controllers}}
 // WARNING: This file was auto-generated with tsoa. Please do not modify it. Re-run tsoa to re-generate this file: https://github.com/lukeautry/tsoa
 import { {{name}} } from '{{modulePath}}';

--- a/packages/cli/src/routeGeneration/templates/hapi.hbs
+++ b/packages/cli/src/routeGeneration/templates/hapi.hbs
@@ -1,7 +1,8 @@
 /* tslint:disable */
 /* eslint-disable */
 // WARNING: This file was auto-generated with tsoa. Please do not modify it. Re-run tsoa to re-generate this file: https://github.com/lukeautry/tsoa
-import { TsoaRoute, fetchMiddlewares, HapiTemplateService } from '@tsoa/runtime';
+import type { TsoaRoute } from '@tsoa/runtime';
+import { fetchMiddlewares, HapiTemplateService } from '@tsoa/runtime';
 {{#each controllers}}
 // WARNING: This file was auto-generated with tsoa. Please do not modify it. Re-run tsoa to re-generate this file: https://github.com/lukeautry/tsoa
 import { {{name}} } from '{{modulePath}}';

--- a/packages/cli/src/routeGeneration/templates/koa.hbs
+++ b/packages/cli/src/routeGeneration/templates/koa.hbs
@@ -1,7 +1,8 @@
 /* tslint:disable */
 /* eslint-disable */
 // WARNING: This file was auto-generated with tsoa. Please do not modify it. Re-run tsoa to re-generate this file: https://github.com/lukeautry/tsoa
-import { TsoaRoute, fetchMiddlewares, KoaTemplateService } from '@tsoa/runtime';
+import type { TsoaRoute } from '@tsoa/runtime';
+import { fetchMiddlewares, KoaTemplateService } from '@tsoa/runtime';
 {{#each controllers}}
 // WARNING: This file was auto-generated with tsoa. Please do not modify it. Re-run tsoa to re-generate this file: https://github.com/lukeautry/tsoa
 import { {{name}} } from '{{modulePath}}';

